shell/key-bindings.bash

```diff
@@ -51,15 +51,16 @@ __fzf_history__() (
   shopt -u nocaseglob nocasematch
   line=$(
     HISTTIMEFORMAT= history |
-    eval "$(__fzfcmd) +s --tac +m -n2..,.. --tiebreak=index --toggle-sort=ctrl-r $FZF_CTRL_R_OPTS" |
-    command grep '^ *[0-9]') &&
-    if [[ $- =~ H ]]; then
-      sed 's/^ *\([0-9]*\)\** .*/!\1/' <<< "$line"
-    else
-      sed 's/^ *\([0-9]*\)\** *//' <<< "$line"
-    fi
+    eval "$(__fzfcmd) +s --tac +m -n2..,.. --tiebreak=index --toggle-sort=ctrl-r $FZF_CTRL_R_OPTS --query=$1" |
+    command grep '^ *[0-9]') && sed 's/^ *\([0-9]*\)\** *//' <<< "$line"
```

Comment:
I was wondering why you removed the conditional branch when it seems unrelated to the suggested change. Anyway, I like the idea. It would be nice if we could implement it with the other shells as well.